### ISSUE TITLE
UNREACHABLE executed at /mnt/e/work/prepo/llvm/lib/IR/RepoHashCalculator.cpp:131!

### ISSUE THREAD

**Issue by gregbedwell** [SNSystems/llvm-prepo#38]
```
$ cat 1.cpp
typedef char a __attribute__((__vector_size__(8)));
void b() { __builtin_ia32_palignr((a)0L, (a)0L, 0); }

$ ./clang++ -c 1.cpp -march=btver2 -target x86_64-pc-gnu-linux
$ ./clang++ -c 1.cpp -march=btver2 -target x86_64-pc-gnu-linux-repo
Unknown type!
UNREACHABLE executed at /mnt/e/work/prepo/llvm/lib/IR/RepoHashCalculator.cpp:131!
#0 0x00007f14fc63d6aa llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x243d6aa)
#1 0x00007f14fc63b2d6 llvm::sys::RunSignalHandlers() (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x243b2d6)
#2 0x00007f14fc63b675 SignalHandler(int) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x243b675)
#3 0x00007f14f9bf2890 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12890)
#4 0x00007f14f887ee97 gsignal (/lib/x86_64-linux-gnu/libc.so.6+0x3ee97)
#5 0x00007f14f8880801 abort (/lib/x86_64-linux-gnu/libc.so.6+0x40801)
#6 0x00007f14fc5dc6ba (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x23dc6ba)
#7 0x00007f14fc170e87 llvm::HashCalculator::hashType(llvm::Type*) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f70e87)
#8 0x00007f14fc173b44 llvm::FunctionHashCalculator::hashInstruction(llvm::Instruction const*) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f73b44)
#9 0x00007f14fc1741a9 llvm::FunctionHashCalculator::hashBasicBlock(llvm::BasicBlock const*) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f741a9)
#10 0x00007f14fc174304 llvm::FunctionHashCalculator::hashFunction() (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f74304)
#11 0x00007f14fc174732 llvm::FunctionHashCalculator::calculateHash() (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f74732)
#12 0x00007f14fc139e72 llvm::DenseMapIterator<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo>, true> llvm::ticketmd::calculateInitialDigestAndDependencies<llvm::Function>(llvm::Function const*, llvm::DenseMap<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo> >&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f39e72)
#13 0x00007f14fc13a7e4 llvm::ticketmd::updateInitialDigestAndGetDependencies(llvm::GlobalObject const*, llvm::MD5&, llvm::DenseMap<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo> >&, llvm::ticketmd::GONumber&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f3a7e4)
#14 0x00007f14fc13b67c void llvm::ticketmd::updateDigestUseCallDependencies<llvm::ticketmd::generateTicketMDs(llvm::Module&)::{lambda(llvm::GlobalObject const*, llvm::MD5&, llvm::DenseMap<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo> >&)#1}>(llvm::GlobalObject const*, llvm::MD5&, llvm::DenseMap<llvm::GlobalObject const*, unsigned int, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, unsigned int> >&, llvm::DenseMap<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo> >&, llvm::ticketmd::generateTicketMDs(llvm::Module&)::{lambda(llvm::GlobalObject const*, llvm::MD5&, llvm::DenseMap<llvm::GlobalObject const*, llvm::ticketmd::GOInfo, llvm::DenseMapInfo<llvm::GlobalObject const*>, llvm::detail::DenseMapPair<llvm::GlobalObject const*, llvm::ticketmd::GOInfo> >&)#1}) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f3b67c)
#15 0x00007f14fc13bf05 llvm::ticketmd::generateTicketMDs(llvm::Module&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1f3bf05)
#16 0x00007f14fc1e8481 (anonymous namespace)::RepoTicketGeneration::runOnModule(llvm::Module&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1fe8481)
#17 0x00007f14fc0f79a1 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x1ef79a1)
#18 0x00007f14fc80dc24 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x260dc24)
#19 0x00007f14fcf69b48 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x2d69b48)
#20 0x00007f14fd38214a clang::ParseAST(clang::Sema&, bool, bool) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x318214a)
#21 0x00007f14fcf665f8 clang::CodeGenAction::ExecuteAction() (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x2d665f8)
#22 0x00007f14fcbef7a6 clang::FrontendAction::Execute() (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x29ef7a6)
#23 0x00007f14fcbb88ce clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x29b88ce)
#24 0x00007f14fcc8b72a clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0x2a8b72a)
#25 0x00007f14faeafff8 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0xcafff8)
#26 0x00007f14fae380dc main (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0xc380dc)
#27 0x00007f14f8861b97 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b97)
#28 0x00007f14faeac24a _start (/mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0+0xcac24a)
Stack dump:
0.      Program arguments: /mnt/e/work/prepo/build-gcc-native-ninja/bin/clang-6.0 -cc1 -triple x86_64-pc-linux-gnu-repo -emit-obj -mrelax-all -disable-free -main-file-name 1.cpp -mrelocation-model static -mthread-model posix -mdisable-fp-elim -fmath-errno -masm-verbose -munwind-tables -fuse-init-array -target-cpu btver2 -dwarf-column-info -debugger-tuning=gdb -coverage-notes-file /mnt/e/work/prepo/build-gcc-native-ninja/bin/1.gcno -resource-dir /mnt/e/work/prepo/build-gcc-native-ninja/lib/clang/6.0.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/x86_64-linux-gnu/c++/7.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/x86_64-linux-gnu/c++/7.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/backward -internal-isystem /usr/local/include -internal-isystem /mnt/e/work/prepo/build-gcc-native-ninja/lib/clang/6.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdeprecated-macro -fdebug-compilation-dir /mnt/e/work/prepo/build-gcc-native-ninja/bin -ferror-limit 19 -fmessage-length 237 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -o 1.o -x c++ 1.cpp
1.      <eof> parser at end of file
2.      Per-module optimization passes
3.      Running pass 'RepoTicketGenerationPass' on module '1.cpp'.
clang-6.0: error: unable to execute command: Aborted (core dumped)
clang-6.0: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 6.0.0 (https://github.com/SNSystems/clang-prepo.git 85663ce2d0e357504b3b92e3cdf53b39056d1f7e) (https://github.com/SNSystems/llvm-prepo.git 1bfded74475ead9e577569905e185799c998863f)
Target: x86_64-pc-linux-gnu-repo
Thread model: posix
InstalledDir: /mnt/e/work/prepo/build-gcc-native-ninja/bin/.
clang-6.0: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang-6.0: note: diagnostic msg:
********************
```

**Comment by MaggieYingYi** [comment]
The issue is caused by a new function return type (‘X86_MMXTyID’). The function return types of ‘HalfTyID’ and ‘X86_MMXTyID’ have been added to function type hash calculator. To catch a new added function return type, the default case of the switch statement is removed as well.

**Comment by MaggieYingYi** [comment]
Fixed in the commit 2b6d671.
